CRM/Contribute/BAO/Contribution.php

```diff
@@ -781,32 +781,45 @@ public static function &importableFields($contactType = 'Individual', $status =
   }
 
   /**
+   * Get exportable fields with pseudoconstants rendered as an extra field.
+   */
+  public static function getExportableFieldsWithPseudoConstants() {
+    $fields = self::exportableFields();
+    foreach ($fields as $field) {
+      if (!empty($field['pseudoconstant']) && !empty($field['pseudoconstant']['optionGroupName'])) {
+        $fields[$field['pseudoconstant']['optionGroupName']] = array(
+          'title' => CRM_Core_BAO_OptionGroup::getTitleByName($field['pseudoconstant']['optionGroupName']),
+          'name' => $field['pseudoconstant']['optionGroupName'],
+          'data_type' => CRM_Utils_Type::T_STRING,
+        );
+      }
+    }
+    return $fields;
+  }
+
+  /**
    * Combine all the exportable fields from the lower level objects.
    *
    * @param bool $checkPermission
    *
    * @return array
    *   array of exportable Fields
    */
-  public static function &exportableFields($checkPermission = TRUE) {
+  public static function exportableFields($checkPermission = TRUE) {
     if (!self::$_exportableFields) {
       if (!self::$_exportableFields) {
         self::$_exportableFields = array();
       }
 
       $impFields = CRM_Contribute_DAO_Contribution::export();
+      foreach ($impFields as &$field) {
+        // @todo consider declaring this at the DAO level.
+        $field['bao'] = 'CRM_Contribute_BAO_Contribution';
+      }
       $expFieldProduct = CRM_Contribute_DAO_Product::export();
       $expFieldsContrib = CRM_Contribute_DAO_ContributionProduct::export();
       $typeField = CRM_Financial_DAO_FinancialType::export();
       $financialAccount = CRM_Financial_DAO_FinancialAccount::export();
-      $optionField = CRM_Core_OptionValue::getFields($mode = 'contribute');
```

Comment:
as part of this PR I declared payment_instrument_id as exportable so it no longer needs manual adding